processing/src/main/java/org/apache/carbondata/processing/newflow/DataLoadProcessorStep.java

```diff
@@ -0,0 +1,40 @@
+package org.apache.carbondata.processing.newflow;
+
+import java.util.Iterator;
+
+import org.apache.carbondata.processing.newflow.exception.CarbonDataLoadingException;
+
+/**
+ * This base interface for data loading. It can do transformation jobs as per the implementation.
+ *
+ */
+public interface DataLoadProcessorStep {
+
+  /**
+   * The output meta for this step. The data returns from this step is as per this meta.
+   * @return
+   */
+  DataField[] getOutput();
+
+  /**
+   * Intialization process for this step.
+   * @param configuration
+   * @param child
+   * @throws CarbonDataLoadingException
+   */
+  void intialize(CarbonDataLoadConfiguration configuration, DataLoadProcessorStep child) throws
+      CarbonDataLoadingException;
+
+  /**
+   * Tranform the data as per the implemetation.
+   * @return Iterator of data
+   * @throws CarbonDataLoadingException
+   */
+  Iterator<Object[]> execute() throws CarbonDataLoadingException;
```

Comment:
I think `execute()` is called for every parallel unit of the input, right? For example, when using spark to load from dataframe, `execute()` is called for every spark partition (execute one task for one partition). When loading from CSV HDFS file, `execute()` is called for every HDFS block. So I do not think returning array of iterator is required.

The loading process of carbon in every executor, some of the step can be parallelized, but sort step need to be synchronized (potential bottle net), since we need datanode-scope sorting. Am I correct?